### PR TITLE
Add ping command to server to allow container healthchecks

### PR DESCRIPTION
**Pull request by xoxys**
Fixes: https://github.com/woodpecker-ci/woodpecker/issues/1943

Note: Kubernetes ignores the container `HEALTHCHECK` by default.

### CODE REVIEW

File: cmd/server/health.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+// handles pinging the endpoint and returns an error if the
+// server is in an unhealthy state.
+func pinger(c *cli.Context) error {
+	serverAddr := c.String("server-addr")
+	if strings.HasPrefix(serverAddr, ":") {
+		// this seems sufficient according to https://pkg.go.dev/net#Dial
+		serverAddr = "localhost" + serverAddr
+	}
+	resp, err := http.Get("http://" + serverAddr + "/healthz")
```
Comment (6543) [inline cmd/server/health.go:34]: can we forward the ping to the database?

**Comment by 6543** [comment]
never mind :laughing:

File: cmd/server/health.go (hunk 1 above)
Comment (6543) [inline cmd/server/health.go:34]: We need to check if woodpecker does ssl itselve and use https in this case instead

**Review by 6543 — CHANGES_REQUESTED** [review]

File: cmd/server/health.go (hunk 1 above)
Comment (xoxys) [reply cmd/server/health.go:34]: That was my question in the initial post...

Comment (xoxys) [reply cmd/server/health.go:34]: Will fix it

Comment (6543) [reply cmd/server/health.go:34]: sure ...

**Review by 6543 — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by xoxys** [comment]
But the ping command is executed as a standalone command. Why should I as a user set let's encrypt config flags etc. on the ping command execution just to get it to use HTTPS. Maybe a more user-friendly way would be something like `woodpecker-server ping --ping-url https://my-ci.example.com`.

If the ping command is executed on the same host in the same context as the server process with access to the same env vars, this might work, but in other situations it might not (even if this might not the desired use case).

**Comment by xoxys** [comment]
However, thanks for finishing it.

**Comment by 6543** [comment]
I dont think we need a option to a custom url.
It should use the same config options as the main server.

users/admins should not use it like the cli tool ...

if we need such a feature, it should be added to the cli tool instead

**Pull request merged**